packages/server-wallet/src/db-admin/db-admin.ts

```diff
@@ -15,53 +16,123 @@ import {LedgerRequest} from '../models/ledger-request';
+   */
+  static async migrateDatabaseFromKnex(knex: Knex): Promise<void> {
     const extensions = [path.extname(__filename)];
-    return this.knex.migrate.latest({
+    return knex.migrate.latest({
       directory: path.join(__dirname, '../db/migrations'),
       loadExtensions: extensions,
     });
   }
 
-  async truncateDB(
-    tables = [
-      SigningWallet.tableName,
-      Channel.tableName,
-      Nonce.tableName,
-      ObjectiveModel.tableName,
-      ObjectiveChannelModel.tableName,
-      Funding.tableName,
-      AppBytecode.tableName,
-      LedgerRequest.tableName,
-      LedgerProposal.tableName,
-      Funding.tableName,
-      AdjudicatorStatusModel.tableName,
-      ChainServiceRequest.tableName,
-    ]
+  /**
+   * Truncates data from all the specified tables
+   * @param config The wallet configuration object containing the database configuration to use
+   * @param tables A list of table names to truncate. Defaults to ALL tables
+   */
+  static async truncateDatabase(
+    config: IncomingServerWalletConfig,
+    tables = defaultTables
   ): Promise<void> {
+    const knex = getKnexFromConfig(config);
+    await DBAdmin.truncateDataBaseFromKnex(knex, tables);
+    knex.destroy();
+  }
+
+  /**
+   * Truncates data from all the specified tables
+   * @param knex A knex instance connected to a wallet database
+   * @param tables A list of table names to truncate, which defaults to ALL tables
+   */
+  static async truncateDataBaseFromKnex(knex: Knex, tables = defaultTables): Promise<void> {
     // eslint-disable-next-line no-process-env
     if (process.env.NODE_ENV !== 'development' && process.env.NODE_ENV !== 'test') {
       throw 'Cannot truncate unless in test or development environments';
     }
-    await Promise.all(tables.map(table => this.knex.raw(`TRUNCATE TABLE ${table} CASCADE;`)));
+    await Promise.all(tables.map(table => knex.raw(`TRUNCATE TABLE ${table} CASCADE;`)));
   }
+}
 
-  get dbName(): string {
-    return this.knex.client.config.connection.database;
-  }
+function getKnexFromConfig(config: IncomingServerWalletConfig): Knex {
+  const populatedConfig = _.assign({}, defaultConfig, config);
+  return Knex(extractDBConfigFromServerWalletConfig(populatedConfig));
+}
+function getDbNameFromKnex(knex: Knex): string {
+  return knex.client.config.connection.database;
 }
+
+function getDbNameFromConfig(config: IncomingServerWalletConfig): string {
+  const populatedConfig = _.assign({}, defaultConfig, config);
+  return getDatabaseConnectionConfig(populatedConfig).database;
+}
+
+const defaultTables = [
+  SigningWallet.tableName,
+  Channel.tableName,
+  Nonce.tableName,
+  ObjectiveModel.tableName,
+  ObjectiveChannelModel.tableName,
+  Funding.tableName,
+  AppBytecode.tableName,
+  LedgerRequest.tableName,
+  LedgerProposal.tableName,
+  Funding.tableName,
+  AdjudicatorStatusModel.tableName,
+  ChainServiceRequest.tableName,
+];
```

Comment:
Something like this might be a bit cleaner
```suggestion
const defaultTables = [
  SigningWallet,
  Channel,
  Nonce,
  ObjectiveModel,
  ObjectiveChannelModel,
  Funding,
  AppBytecode,
  LedgerRequest,
  LedgerProposal,
  Funding,
  AdjudicatorStatusModel,
  ChainServiceRequest,
].map(model => model.tableName);
```